src/Orleans.Core.Abstractions/IDs/ClientGrainId.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// Identifies a client.
+    /// </summary>
+    internal readonly struct ClientGrainId : IEquatable<ClientGrainId>, IComparable<ClientGrainId>
+    {
+        /// <summary>
+        /// Creates a new <see cref="ClientGrainId"/> instance.
+        /// </summary>
+        private ClientGrainId(GrainId grainId) => this.GrainId = grainId;
+
+        /// <summary>
+        /// Gets the underlying <see cref="GrainId"/>.
+        /// </summary>
+        public GrainId GrainId { get; }
+
+        /// <summary>
+        /// Creates a new <see cref="ClientGrainId"/> instance.
+        /// </summary>
+        public static ClientGrainId Create() => Create(Guid.NewGuid().ToString("N"));
+
+        /// <summary>
+        /// Creates a new <see cref="ClientGrainId"/> instance.
+        /// </summary>
+        public static ClientGrainId Create(string id) => Create(IdSpan.Create(id));
```

Comment:
Do we plan to enable the client name to be configurable? It might be dangerous.

Reply:
I'm not sure. It can be useful for the hosted client case.